### DISCUSSION TITLE
能否添加 端口规则 --关于force_dnscrypt_list的补充

### DISCUSSION THREAD

**Discussion by falltree** [kkkgo/PaoPaoDNS#204] [想法]
目前开了cnauto=yes后，有一些特定软件比如zeotier内网穿透，它们使用固定的端口9993，需要它们直连，而不走代理，这时如果能够以端口9993为关键词，强制其走force_dnscrypt，就更加方便了。

zerotier只是一个例子，可能还有其他类似的网站或服务是某个特定的端口，希望能通过以端口为规则，强制它们走custom forward或者force dnscrypt

**Comment by kkkgo** [comment]
zeotier就没有DNS解析的过程，他是IP硬编码的。
另外需要补充下网络基础知识，DNS域名没有端口的说法。

**Reply by falltree** [reply]
我不是专业人士，从来就没有专门学过网络知识，只是从使用角度举一个例子。
现在开启cnauto后，有一些不需要代理的都走了custom forward，从而流量都从代理走了。
而目前可设置的规则只有域名，关键词，所以觉得paopaodns或许可以增加一个端口的识别，比如访问http://我的自建境外服务器:9999，我需要它直连，而不是从代理走，当有几个类似的，不同域名的服务器都是使用这个9999端口时，那么我只需要针对9999这个规则走fore dnscrpt，就不需要一条一条添加域名关键词了。

**Reply by kkkgo** [reply]
DNS没有端口的说法=不能使用端口来分流。
这个分流和你用的傻瓜代理软件原理上就不一样，代理的流量根本不经过dns docker，我怎么知道你访问了什么端口呢？
另一方面，能分流过去的就肯定是能用域名控制的，无论你是怎么样的服务器，或者怎么样的通配域名。

**Reply by kkkgo** [reply]
你真要实现这种需求，只能是在代理网关上做，虽然给网关加规则有点多此一举，但只有网关知道你在访问什么端口。

**Reply by falltree** [reply]
ok，看来dns是做不了这个，那就只能在网关设置了。这种需求对于大多数人当然是不需要的，但是每个人都有自己特定的情况。
另外请教一下在ros container添加paopaodns，
 /container/add remote-image=sliamb/paopaodns:latest interface=veth1 root-dir=docker/paopaodns logging=yes mounts=dns  envlist=dns start-on-boot=yes comment=paopaodns

这种情况下貌似只有默认的53被映射了，运行test会提示5304 failed。
该如何添加port 5304还有7889，直接在ros container的env list？这里好像都是环境变量，似乎无法添加port
![image](https://github.com/user-attachments/assets/b90f78ea-06e0-4f8a-9d5a-e51c17865a04)

**Reply by kkkgo** [reply]
我没用过ros container。建议使用标准容器环境docker compose部署。
